### DISCUSSION TITLE
Enabling preserved CSS modules in output

### DISCUSSION THREAD

**Discussion by kirkobyte** [vanilla-extract-css/vanilla-extract#367] [Ideas]
It would be great to allow preserving modules in the CSS output by this tool.

In the case of building a design system, it can be good to leave it up to consuming applications to bundle CSS so they can tree-shake and code-split as their app requires. E.g. if a user was to import `Component` from a design system, they wouldn't also want the CSS from `Component2`, so separate CSS files could be useful.
```
src/
├── components
│   ├── Component
│   │  ├ index.tsx
│   │  └ index.css.ts
│   └── // ...
└── // ...
```
```
dist/
├── components
│   ├── Component
│   │  ├ index.js
│   │  └ index.css
│   └── // ...
└── // ...
```

In [a previous discussion around supporting other bundlers](https://github.com/seek-oss/vanilla-extract/issues/3) in this repo, it was stated that Rollup may not be a good target for `vanilla-extract` at this point because it doesn't have first-class CSS bundling support:
> Just an update on the Rollup integration work—since it doesn’t have first class CSS bundling support, we're thinking it’s probably not a good target for static extraction.

If this use-case is something Vanilla Extract would like to support, either enabling support for `Rollup` with `preserveModules: true` or adding support for preserving separate CSS modules in the ESBuild plugin could be really promising

**Comment by Piero87** [comment]
I'm looking for an answer about this topic too
